server/src/managers/websocket/connection.rs

```diff
@@ -373,10 +533,56 @@ pub(crate) mod test {
         assert!(req.body.unwrap() == env.encode_to_vec());
     }
 
-    #[ignore = "Not implemented"]
     #[tokio::test]
-    async fn test_on_receive() {
-        todo!()
+    async fn test_on_receive_request() {
+        let mut state = SignalServerState::<MockDB, MockSocket>::new();
+        let (mut client, sender, mut receiver, mut mreceiver) =
+            create_connection("a", 1, "127.0.0.1:4042", state);
+        let msg = r#"
+        {
+            "messages":[
+                {
+                    "type": 1,
+                    "destinationDeviceId": 3,
+                    "destinationRegistrationId": 22,
+                    "content": "aGVsbG8="
+                }
+            ],
+            "online": false,
+            "urgent": true,
+            "timestamp": 1730217386
+        }
+        "#
+        .as_bytes()
+        .to_vec();
+
+        client
+            .on_receive(create_request(
+                1,
+                "PUT",
+                &format!("/v1/messages/{}", client.protocol_address().name()),
+                vec![],
+                Some(msg),
+            ))
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_on_receive_response() {
```

Comment:
add a test that tests that one client can send a message to another, by inserting two clients into the websocketmanager and having alice's sender send a message and then bobs receiver receive a message after a short sleep